listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -666,6 +677,155 @@ def dump_listens(self, location, dump_id, start_time=datetime.utcfromtimestamp(0
+        """
+            Carry out fetching listens from the DB, joining them to the MBID mapping table and
+            then writing them to parquet files.
+
+        Args:
+            archive_dir: the directory where the listens dump archive should be created
+            tmp_dir: the directory where tmp files should be written
+            dump_id (int): the ID of the dump in the dump sequence
+            start_time and end_time (datetime): the time range for which listens should be dumped
+                start_time defaults to utc 0 (meaning a full dump) and end_time defaults to the current time
+            full_dump (bool): Is this a full or incremental dump?
+        """
+
+        listen_count = 0
+
+        if start_time:
+            start_time = datetime.utcfromtimestamp(datetime.timestamp(start_time))
+        if end_time:
+            end_time = datetime.utcfromtimestamp(datetime.timestamp(end_time))
+        else:
+            end_time = datetime.now()
+
+        parquet_file_id = 0
+
+        query = """SELECT listened_at,
+                          user_name,
+                          data->'track_metadata'->>'artist_name' AS artist_name,
+                          artist_credit_id,
+                          data->'track_metadata'->>'release_name' AS release_name,
+                          release_mbid::TEXT,
+                          track_name AS recording_name,
+                          recording_mbid::TEXT,
+                          created
+                     FROM listen l
+                     JOIN listen_mbid_mapping m
+                       ON (data->'track_metadata'->'additional_info'->>'recording_msid')::uuid = recording_msid
+                    WHERE created > %s
+                      AND created <= %s
+                      AND recording_mbid IS NOT NULL
+                 ORDER BY created ASC"""
+
+        args = (start_time, end_time)
+
+        listen_count = 0
+
+        current_created = None
+        conn = timescale.engine.raw_connection()
+        with conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor) as curs:
+            curs.execute(query, args)
+            while True:
+                t0 = time.monotonic()
+                written = 0
+                approx_size = 0
+                data = {
+                    'listened_at': [],
+                    'user_name': [],
+                    'artist_name': [],
+                    'artist_credit_id': [],
+                    'release_name': [],
+                    'release_mbid': [],
+                    'recording_name': [],
+                    'recording_mbid': [],
+                    'created': []
+                }
+                while True:
+                    result = curs.fetchone()
+                    if not result:
+                        break
+
+                    for col in data:
+                        if col == 'artist_credit_id':
+                            data[col].append(int(result[col]))
```

Comment:
Can we cast the the column in the sql query instead?

Reply:
The column is an int -- I am not even certain why this is necessary; by default it goes to double as you saw.

Reply:
Right, that's why its weird.